types/errors/errors_test.go

```diff
@@ -155,56 +155,95 @@ func (customError) Error() string {
 	require.True(t, stdlib.Is(err, ErrTxTooLarge))
 
-	err = Wrap(err, "more context")
+	err = Extend(err, "more context")
 	require.True(t, stdlib.Is(err, ErrTxTooLarge))
 
-	err = Wrap(err, "even more context")
+	err = Extend(err, "even more context")
 	require.True(t, stdlib.Is(err, ErrTxTooLarge))
 
-	err = Wrap(ErrInsufficientFee, "...")
+	err = Extend(ErrInsufficientFee, "...")
 	require.False(t, stdlib.Is(err, ErrTxTooLarge))
 }
 
-func TestWrappedIsMultiple(t *testing.T) {
+func TestExtendedIsMultiple(t *testing.T) {
 	var errTest = errors.New("test error")
 	var errTest2 = errors.New("test error 2")
-	err := Wrap(errTest2, Wrap(errTest, "some random description").Error())
+	err := Extend(errTest2, Extend(errTest, "some random description").Error())
 	require.True(t, stdlib.Is(err, errTest2))
 }
 
-func TestWrappedIsFail(t *testing.T) {
+func TestExtendedIsFail(t *testing.T) {
 	var errTest = errors.New("test error")
 	var errTest2 = errors.New("test error 2")
-	err := Wrap(errTest2, Wrap(errTest, "some random description").Error())
+	err := Extend(errTest2, Extend(errTest, "some random description").Error())
 	require.False(t, stdlib.Is(err, errTest))
 }
 
-func TestWrappedUnwrap(t *testing.T) {
+func TestExtendedUnwrap(t *testing.T) {
 	var errTest = errors.New("test error")
-	err := Wrap(errTest, "some random description")
+	err := Extend(errTest, "some random description")
 	require.Equal(t, errTest, stdlib.Unwrap(err))
 }
 
-func TestWrappedUnwrapMultiple(t *testing.T) {
+func TestExtendedUnwrapMultiple(t *testing.T) {
 	var errTest = errors.New("test error")
 	var errTest2 = errors.New("test error 2")
-	err := Wrap(errTest2, Wrap(errTest, "some random description").Error())
+	err := Extend(errTest2, Extend(errTest, "some random description").Error())
 	require.Equal(t, errTest2, stdlib.Unwrap(err))
 }
 
-func TestWrappedUnwrapFail(t *testing.T) {
+func TestExtendedUnwrapFail(t *testing.T) {
 	var errTest = errors.New("test error")
 	var errTest2 = errors.New("test error 2")
-	err := Wrap(errTest2, Wrap(errTest, "some random description").Error())
+	err := Extend(errTest2, Extend(errTest, "some random description").Error())
 	require.NotEqual(t, errTest, stdlib.Unwrap(err))
 }
+
+func TestExtendf(t *testing.T) {
+	tests := []struct {
+		name          string
+		err           error
+		format        string
+		args          []interface{}
+		wantNilErr    bool
+		wantErrString string
+	}{
+		{"nil", nil, "", []interface{}{}, true, ""},
+		{"no format", errors.New("error"), "", []interface{}{}, false, "error: "},
+		{"format and args", errors.New("error"), "%s %d", []interface{}{"code", -1}, false, "error: code -1"},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			err := Extendf(tt.err, tt.format, tt.args...)
+			if tt.wantNilErr {
+				require.Nil(t, err)
+				return
+			}
+			require.Equal(t, tt.wantErrString, err.Error())
+		})
+	}
+}
+
+func ExampleExtend() {
```

Comment:
nit, these can be made private

Reply:
These are testable examples: https://blog.golang.org/examples

Reply:
If we made it private, it would probably not show up in the go docs